paimon-core/src/main/java/org/apache/paimon/catalog/AbstractCatalog.java

```diff
@@ -54,32 +61,43 @@
 import static org.apache.paimon.utils.Preconditions.checkArgument;
 
 /** Common implementation of {@link Catalog}. */
-public abstract class AbstractCatalog implements Catalog {
+public abstract class AbstractCatalog implements Catalog, PrivilegedCatalog {
```

Comment:
Maybe we should wrap a catalog just like `PrevilegedFileStoreTable`.

Reply:
Some catalogs may not use FileSystem based Previlege system.
For example, RestCatalog will register users in server not filesystem.